### ISSUE TITLE
Elided text causes line numbers to overlap

### ISSUE THREAD

**Issue by Akuli** [Moosems/TkLineNums#15]
```
from platform import system
from tkinter import Text, Tk
from tkinter.ttk import Style

from tklinenums import TkLineNumbers

# This is to make the example work on both Windows and Mac
if system() == "Darwin":
    contmand: str = "Command"
else:
    contmand: str = "Control"

# Create the root window
root = Tk()

# Set the ttk style (tkinter's way of styling) for the line numbers
style = Style()
style.configure("TLineNumbers", background="#ffffff", foreground="#2197db")

# Create the Text widget and pack it to the right
text = Text(root)
text.pack(side="right")

# Insert 50 lines of text into the Text widget
for i in range(100):
    text.insert("end", f"Line {i+1}\n")

text.tag_config("foo", elide=True)
text.tag_add("foo", "20.0", "80.0")

# Create the TkLineNumbers widget and pack it to the left
linenums = TkLineNumbers(root, text)
linenums.pack(fill="y", side="left")

# Create binds to redraw the line numbers for most common events that change the text in the Text widget
text.bind("<Key>", lambda event: root.after_idle(linenums.redraw), add=True)
text.bind("<BackSpace>", lambda event: root.after_idle(linenums.redraw), add=True)
text.bind(f"<{contmand}-v>", lambda event: root.after_idle(linenums.redraw), add=True)

# Start the mainloop for the root window
root.mainloop()
```

![s](https://user-images.githubusercontent.com/18505570/233803074-dbaa997b-adbe-4cc2-818c-30982268bd40.png)

**Comment by Akuli** [comment]
Porcupine's solution:

```python3
def line_is_elided(textwidget: tkinter.Text, lineno: int) -> bool:
    tags = textwidget.tag_names(f"{lineno}.0")
    elide_values = (textwidget.tag_cget(tag, "elide") for tag in tags)
    # elide values can be empty
    return any(tkinter.getboolean(v or "false") for v in elide_values)

...

        for lineno in range(first_line, last_line + 1):
            # index('@0,y') doesn't work when scrolled a lot to side, but dlineinfo seems to work
            dlineinfo = self._textwidget.dlineinfo(f"{lineno}.0")
            if dlineinfo is None or line_is_elided(self._textwidget, lineno):
                # line not on screen for whatever reason
                continue
```

Presumably this needs something more advanced if only a part of a line can be elided. Maybe we should check whether the newline characters at the ends of the lines are elided.

**Comment by Moosems** [comment]
Fixed over in the `click-drag-bugs` branch. If you could help me to fix them I can make a `1.7.0` release to add all your features :D
